addons/mail/models/mail_activity.py

```diff
@@ -631,6 +643,20 @@ def _search_activity_user_id(self, operator, operand):
     def _search_activity_type_id(self, operator, operand):
         return [('activity_ids.activity_type_id', operator, operand)]
 
+    @api.multi
+    def _set_exception_activity_type(self):
+        self.ensure_one()
+        decoration_type = None
+        icon = None
+        for activity in self.activity_ids.filtered(lambda x: x.activity_type_id.decoration_type in ['warning', 'danger']):
+            if not decoration_type and activity.activity_type_id.decoration_type == 'warning':
+                decoration_type = 'warning'
+                icon = activity.activity_type_id.icon
+            elif activity.activity_type_id.decoration_type == 'danger':
+                decoration_type = 'danger'
+                icon = activity.activity_type_id.icon
+                break
+        self.exception_activity_type = "pull-right text-%s fa %s" % (decoration_type, icon)
```

Comment:
I have to admit that I don't like this server-side formatting at all. This should be the role of the client (deciding how to display and which classes apply).

Possible solutions:
 * still keep one char field `warning,fa-tasks`, and the widget splits the value. This is still not perfect because one field has to carry multiple information, and you may have inconsistency
 * two fields 1) field containing `warning` (might be a selection instead) and 2) second field containg`fa-tasks` ; the JS widget would be on one of them (I'd go for the first field, because this is the one making sense, the second one is more like a complement of information) and the JS widget could use `fieldDependencies` property (see https://github.com/odoo/odoo/blob/master/addons/web/static/src/js/fields/abstract_field.js#L50 so you don't have to add the field in invisible in every view) to fetch the second field ; this proposition seems to be the best according to me but performance-wise, one might check if the `fieldDependencies` are correctly read in batch.

Reply:
Oh, I check this morning and this is the case so we should probably go for the second solution.